File: assignments/ruby/robot-name/robot-name_test.rb
require 'minitest/autorun'
require 'minitest/pride'
require_relative 'robot'

class RobotTest < MiniTest::Unit::TestCase
  def test_has_name
    assert_match /\w{2}\d{3}/, Robot.new.name
  end

  def test_name_sticks
    robot = Robot.new
    robot.name
    assert_equal robot.name, robot.name
  end

  def test_different_robots_have_different_names
    assert Robot.new.name != Robot.new.name
  end

  def test_reset_name
    robot = Robot.new
    name = robot.name
    robot.reset
    name2 = robot.name
    assert name != name2
    assert_match /\w{2}\d{3}/, name2
  end
end

File: assignments/python/robot-name/robot_name_test.py
try:
    from robot import Robot
except ImportError:
    raise SystemExit('Could not find robot.py. Does it exist?')

import unittest


class RobotTest(unittest.TestCase):
    name_re = r'\w{2}\d{3}'

    def test_has_name(self):
        self.assertRegexpMatches(Robot().name, self.name_re)

    def test_name_sticks(self):
        robot = Robot()
        robot.name
        self.assertEqual(robot.name, robot.name)

    def test_different_robots_have_different_names(self):
        self.assertNotEqual(
            Robot().name,
            Robot().name
        )

    def test_rest_name(self):
        robot = Robot()
        name = robot.name
        robot.reset()
        name2 = robot.name
        self.assertNotEqual(name, name2)
        self.assertRegexpMatches(name2, self.name_re)

if __name__ == '__main__':
    unittest.main()
